### PR TITLE
Merge Release 1.4 into develop

### PR DESCRIPTION
**Pull request by boblangley**
Cherry picked commits that were applied to master for 1.4

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@boblangley there's a R# inspection error. Once it's sorted out, good to merge.

File: src/NServiceBus.Persistence.AzureStorage/SagaPersisters/SecondaryIndices/SecondaryIndexPersister.cs [hunk 1]
```diff
@@ -12,11 +12,12 @@ class SecondaryIndexPersister
     {
         public delegate Task<Guid[]> ScanForSagas(Type sagaType, string propertyName, object propertyValue);
 
-        public SecondaryIndexPersister(Func<Type, Task<CloudTable>> getTableForSaga, ScanForSagas scanner, Func<IContainSagaData, PartitionRowKeyTuple?, ContextBag, Task> persist)
+        public SecondaryIndexPersister(Func<Type, Task<CloudTable>> getTableForSaga, ScanForSagas scanner, Func<IContainSagaData, PartitionRowKeyTuple?, ContextBag, Task> persist, bool assumeSecondaryIndcisExist)
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureStorage/SagaPersisters/SecondaryIndices/SecondaryIndexPersister.cs]: >assumeSecondaryIndcisExist

typo

assumeSecondary**Indices**Exist

File: src/NServiceBus.Persistence.AzureStorage/SagaPersisters/SecondaryIndices/SecondaryIndexPersister.cs [hunk 2]
```diff
@@ -237,11 +243,9 @@ static bool IsConflict(StorageException ex)
         }
 
         LRUCache<PartitionRowKeyTuple, Guid> cache = new LRUCache<PartitionRowKeyTuple, Guid>(LRUCapacity);
-
         Func<Type, Task<CloudTable>> getTableForSaga;
-
         Func<IContainSagaData, PartitionRowKeyTuple?, ContextBag, Task> persist;
-
+        bool assumeSecondaryIndcisExist;
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureStorage/SagaPersisters/SecondaryIndices/SecondaryIndexPersister.cs]: same as above

**Comment by SeanFeldman** [comment]
@boblangley needs to be rebased + a tiny typo [here](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/156#pullrequestreview-59386992)

**Comment by boblangley** [comment]
Unable to rebase. Withdrawing this PR as the changes from 1.4 will have to be applied by hand after the merge of #157

**Comment by SeanFeldman** [comment]
Cherry picked `release-1.4` commits manually and committed to `develop`.

**Pull request closed without merging**